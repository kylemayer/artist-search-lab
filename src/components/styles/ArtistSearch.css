* {
  background-color: #394879;
}

.page {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

  ul {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
    font-family: 'Operator mono', sans-serif;
  }

  ul > li {
    display: flex;
    justify-content: center;
    width: fit-content;
    border: 1px solid #c9c9c9;
    border-radius: 3px;
    font-size: 20px;
    background-color: #7e6382;
    padding: 6px;
  }

  a {
    text-decoration: none;
    color: white;
    font-size: 18px;
    background-color: #7e6382;
  }

label {
    font-size: 20px;
    padding: 5px;
    margin: 5px;
    background-color: #7e6382;
  }

  input {
    margin: 5px;
  }

  h3 {
    background-color: #7e6382;
  }

  li {
    background-color: #7e6382;
    margin: 5px;
  }