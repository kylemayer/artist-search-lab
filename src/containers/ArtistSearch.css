.page {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

  ul {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
    font-family: 'Operator mono', sans-serif;
  }

  ul > li {
    display: flex;
    justify-content: center;
    width: fit-content;
    border: 1px solid #c9c9c9;
    border-radius: 3px;
    font-size: 20px;
    background-color: #7e6382;
    padding: 6px;
  }

  a {
    text-decoration: none;
    color: white;
    font-size: 18px;
  }
