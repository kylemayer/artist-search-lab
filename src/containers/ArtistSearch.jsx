import React, { useState, useEffect } from 'react';
import { getArtists } from '../services/fetchApi.js';
import Search from '../components/controls/Search';
import ArtistList from '../components/displays/ArtistList';
import Pagination from 'react-js-pagination';

const ArtistSearch = () => {
  const [loading, setLoading] = useState(true);
  const [artists, setArtists] = useState([]);
  const [searchTerm, setSearchTerm] = useState('');
  const [activePage, setActivePage] = useState(1);

  useEffect(() => {
    const loadArtists = async () => {
      if(!searchTerm) return setLoading(false);
      const artistsFromApi = await getArtists(searchTerm);
      setArtists(artistsFromApi);
      setLoading(false);
    };

    loadArtists();
  }, [searchTerm]);

  const handleSearch = ({ target }) => {
    setSearchTerm(target.value);
  };

  const handlePageChange = (pageNumber) => {
    setActivePage(pageNumber);
  };

  if (loading) return <h3>Loading...</h3>;

  return (
    <>
      <Search searchTerm={searchTerm} onSearch={handleSearch} />
      <Pagination 
        activePage={activePage}
        totalItemsCount={666}
        onChange={handlePageChange}
      />
      <ArtistList
        artists={artists}
        searchTerm={searchTerm}
      />
    </>
  );
};

export default ArtistSearch;
