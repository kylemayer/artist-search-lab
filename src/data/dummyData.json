{
    "created": "2021-09-24T17:48:52.178Z",
    "count": 5,
    "offset": 0,
    "artists": [
        {
            "id": "2339bc21-aa92-4850-86f0-4bb9433910c8",
            "type": "Group",
            "type-id": "e431f5f6-b5d2-343d-8b36-72607fffb74b",
            "score": 100,
            "name": "Butthole Surfers",
            "sort-name": "Butthole Surfers",
            "country": "US",
            "area": {
                "id": "489ce91b-6658-3307-9877-795b68554c98",
                "type": "Country",
                "type-id": "06dd0ae4-8c74-30bb-b43d-95dcedf961de",
                "name": "United States",
                "sort-name": "United States",
                "life-span": {
                    "ended": null
                }
            },
            "begin-area": {
                "id": "a6f7157a-bfab-49e8-a22b-240ade4552ca",
                "type": "City",
                "type-id": "6fd8f29a-3d0a-32fc-980d-ea697b69da78",
                "name": "San Antonio",
                "sort-name": "San Antonio",
                "life-span": {
                    "ended": null
                }
            },
            "isnis": [
                "0000000123593310"
            ],
            "life-span": {
                "begin": "1982",
                "ended": null
            },
            "aliases": [
                {
                    "sort-name": "BH Surfers",
                    "name": "BH Surfers",
                    "locale": null,
                    "type": null,
                    "primary": null,
                    "begin-date": null,
                    "end-date": null
                },
                {
                    "sort-name": "Butthole Sufers",
                    "name": "Butthole Sufers",
                    "locale": null,
                    "type": null,
                    "primary": null,
                    "begin-date": null,
                    "end-date": null
                }
            ],
            "tags": [
                {
                    "count": 1,
                    "name": "alternative rock"
                },
                {
                    "count": 1,
                    "name": "punk"
                },
                {
                    "count": 1,
                    "name": "alternative"
                },
                {
                    "count": 1,
                    "name": "psychedelic rock"
                },
                {
                    "count": 1,
                    "name": "experimental rock"
                },
                {
                    "count": 1,
                    "name": "art punk"
                },
                {
                    "count": 1,
                    "name": "noise rock"
                }
            ]
        },
        {
            "id": "bf6610b0-4d03-49ef-af36-76602eb859de",
            "score": 89,
            "name": "Butthole Cruisers",
            "sort-name": "Butthole Cruisers",
            "life-span": {
                "ended": null
            }
        },
        {
            "id": "1d9d1856-aef9-4a2c-968d-fcd323a119c4",
            "type": "Group",
            "type-id": "e431f5f6-b5d2-343d-8b36-72607fffb74b",
            "score": 89,
            "name": "Butthole Blowhole",
            "sort-name": "Butthole Blowhole",
            "country": "US",
            "area": {
                "id": "489ce91b-6658-3307-9877-795b68554c98",
                "type": "Country",
                "type-id": "06dd0ae4-8c74-30bb-b43d-95dcedf961de",
                "name": "United States",
                "sort-name": "United States",
                "life-span": {
                    "ended": null
                }
            },
            "disambiguation": "goregrind",
            "life-span": {
                "ended": null
            }
        },
        {
            "id": "fd3e84c8-0ebd-4fad-8396-35644c29e6fc",
            "type": "Person",
            "type-id": "b6e035f4-3ce9-331c-97df-83397230b0df",
            "score": 49,
            "name": "Brendan",
            "sort-name": "Brendan",
            "disambiguation": "member of Butthole Blowhole",
            "life-span": {
                "ended": null
            }
        },
        {
            "id": "87035eaf-5886-48c5-ab2f-554edcf2fadf",
            "type": "Group",
            "type-id": "e431f5f6-b5d2-343d-8b36-72607fffb74b",
            "score": 48,
            "name": "Multiple Buttholes",
            "sort-name": "Multiple Buttholes",
            "area": {
                "id": "76d1e078-7861-4329-8553-f047b922c4f7",
                "type": "City",
                "type-id": "6fd8f29a-3d0a-32fc-980d-ea697b69da78",
                "name": "Blacksburg",
                "sort-name": "Blacksburg",
                "life-span": {
                    "ended": null
                }
            },
            "disambiguation": "Blacksburg, VA Band",
            "life-span": {
                "ended": null
            }
        }
    ]
}