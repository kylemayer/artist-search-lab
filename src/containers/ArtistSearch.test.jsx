import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import ArtistSearch from './ArtistSearch';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import dummyData from '../data/dummyData';


const mockServer = setupServer(
  rest.get('http://musicbrainz.org/ws/2/artist', (req, res, ctx) => {
    return res(ctx.json(dummyData));
  })
);

describe('ArtistSearch', () => {
  beforeAll(() => mockServer.listen());
  afterAll(() => mockServer.close());

  it('should display a list of asrtists that match the searched term', async () => {
    const { container } = render(
      <MemoryRouter>
        <ArtistSearch />
      </MemoryRouter>
    );

    const searchBar = await screen.findByRole('searchBar', { name: 'searchBar' });
    userEvent.type(searchBar, '{selectall}{del}butthole');

    return waitFor(() => {
      const list = screen.getAllByText('butthole', { exact: false });

      expect(list.length).toBeGreaterThan(1);
      expect(container).toMatchSnapshot();
    });
  });
});
