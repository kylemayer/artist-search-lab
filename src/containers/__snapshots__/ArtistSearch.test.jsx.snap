// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistSearch should display a list of asrtists that match the searched term 1`] = `
<div>
  <div
    class="artistsearch"
  >
    <label>
      Search for an artist
      <input
        aria-label="searchBar"
        name="searchTerm"
        role="searchBar"
        type="text"
        value="butthole"
      />
    </label>
    <button
      disabled=""
    >
      ←
    </button>
    <button
      disabled=""
    >
      →
    </button>
    <ul
      aria-label="artists"
    >
      <li>
        <a
          href="/artists/Butthole Surfers/2339bc21-aa92-4850-86f0-4bb9433910c8"
        >
          <figure>
            <figcaption>
              <h3>
                Butthole Surfers
              </h3>
            </figcaption>
          </figure>
        </a>
      </li>
      <li>
        <a
          href="/artists/Butthole Cruisers/bf6610b0-4d03-49ef-af36-76602eb859de"
        >
          <figure>
            <figcaption>
              <h3>
                Butthole Cruisers
              </h3>
            </figcaption>
          </figure>
        </a>
      </li>
      <li>
        <a
          href="/artists/Butthole Blowhole/1d9d1856-aef9-4a2c-968d-fcd323a119c4"
        >
          <figure>
            <figcaption>
              <h3>
                Butthole Blowhole
              </h3>
            </figcaption>
          </figure>
        </a>
      </li>
      <li>
        <a
          href="/artists/Brendan/fd3e84c8-0ebd-4fad-8396-35644c29e6fc"
        >
          <figure>
            <figcaption>
              <h3>
                Brendan
              </h3>
            </figcaption>
          </figure>
        </a>
      </li>
      <li>
        <a
          href="/artists/Multiple Buttholes/87035eaf-5886-48c5-ab2f-554edcf2fadf"
        >
          <figure>
            <figcaption>
              <h3>
                Multiple Buttholes
              </h3>
            </figcaption>
          </figure>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
